### PR TITLE
gather-drop improvements

### PR DESCRIPTION
**Pull request by mmitche**
Contains a variety of improvements to gather-drop that focus on how the tool is normally used. The typical scenario is that someone runs gather drop to get a unified drop, the again to get the release layout. This is expensive, order dependent, and clunky. Instead, this commit changes the tool to gather both drops at the same time. In addition:
- Always use json for the manifest file (the parameter is not required)
- Call out a few commits in repos that do not have builds (because they pre-date maestro). Ignore these when determining whether there are dependencies missing builds. This is so we can use --continue-on-error.
- Trim the name, commit, and version parameters in version details files. This was causing the previous improvement to not work for a specific node.
- Copy files to both the release layout and the unified layout in one pass. This reduces the number of downloads.
- Always generate both layouts
- Remove the release layout parameter

### CODE REVIEW

**Comment by mmitche** [comment]
For https://github.com/dotnet/arcade/issues/4983

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs [hunk 1]
```diff
@@ -721,6 +667,12 @@ private IEnumerable<Build> FilterReleasedBuilds(IEnumerable<Build> builds)
             return builds;
         }
 
+        private static readonly List<(string repo, string sha)> DependenciesAlwaysMissingBuilds = new List<(string, string)>()
```
Comment (riarenas) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs]: Should this be a field on top of the file?

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs [hunk 2]
```diff
@@ -1636,29 +1641,37 @@ private static bool IsAzureDevOpsArtifactsUrl(string location)
 
                         using (var inStream = await response.Content.ReadAsStreamAsync())
                         {
-                            downloadOutput.Append($"  {sourceUri} => {targetFile}...");
+                            downloadOutput.AppendLine($"    {sourceUri} =>");
+                            foreach (string targetFile in targetFiles)
+                            {
+                                downloadOutput.AppendLine($"      {targetFile} =>");
```
Comment (riarenas) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs]: What is the => in this one pointing to?
This looks like it'll be something like

    sourceUri =>
        target1 =>
        target2 =>
        tagetn =>

Comment (mmitche) [reply src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs]: From source uri to each output target file locally.

Comment (riarenas) [reply src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs]: Yeah, I mean the arrows at the end of each local file, they doesn't seem to be pointing anywhere.

Comment (mmitche) [reply src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs]: Oh, duh.

**Pull request merged**